sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    };
+    workspaceName?: string;
+}
+
+// @public
+export interface SparkBatchJobCollection {
+    from: number;
+    sessions?: SparkBatchJob[];
+    total: number;
+}
+
+// @public (undocumented)
+export interface SparkBatchJobOptions {
+    // (undocumented)
+    archives?: string[];
+    // (undocumented)
+    arguments?: string[];
+    // (undocumented)
+    artifactId?: string;
+    // (undocumented)
+    className?: string;
+    configuration?: {
+        [propertyName: string]: string;
+    };
+    // (undocumented)
+    driverCores?: number;
+    // (undocumented)
+    driverMemory?: string;
+    // (undocumented)
+    executorCores?: number;
+    // (undocumented)
+    executorCount?: number;
+    // (undocumented)
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
+    // (undocumented)
+    files?: string[];
+    // (undocumented)
+    jars?: string[];
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    pythonFiles?: string[];
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export type SparkBatchJobResultType = "Uncertain" | "Succeeded" | "Failed" | "Cancelled" | string;
+
+// @public (undocumented)
+export interface SparkBatchJobState {
+    currentState?: string;
+    deadAt?: Date;
+    // (undocumented)
+    jobCreationRequest?: SparkRequest;
+    notStartedAt?: Date;
+    recoveringAt?: Date;
+    runningAt?: Date;
+    startingAt?: Date;
+    successAt?: Date;
+    terminatedAt?: Date;
+}
+
+// @public (undocumented)
+export class SparkClient {
+    constructor(workspaceEndpoint: string, sparkPoolName: string, credential: TokenCredential, pipelineOptions?: SparkClientOptions);
+    cancelSparkBatchJob(batchId: number, options?: CancelSparkBatchJobOptions): Promise<OperationResponse>;
+    cancelSparkSession(sessionId: number, options?: CancelSparkSessionOptions): Promise<OperationResponse>;
+    cancelSparkStatement(sessionId: number, statementId: number, options?: CancelSparkStatementOptions): Promise<OperationResponse>;
+    createSparkBatchJob(sparkBatchJobOptions: SparkBatchJobOptions, options?: CreateSparkBatchJobOptions): Promise<CreateSparkBatchJobResponse>;
+    createSparkSeesion(sparkSessionOptions: SparkSessionOptions, options?: CreateSparkSessionOptions): Promise<CreateSparkSessionResponse>;
+    createSparkStatement(sessionId: number, sparkStatementOptions: SparkStatementOptions, options?: CreateSparkStatementOptions): Promise<CreateSparkStatementResponse>;
+    getSparkBatchJob(batchId: number, options?: GetSparkBatchJobOptions): Promise<GetSparkBatchJobResponse>;
+    getSparkSession(sessionId: number, options?: GetSparkSessionOptions): Promise<GetSparkSessionResponse>;
+    getSparkStatement(sessionId: number, statementId: number, options?: GetSparkStatementOptions): Promise<GetSparkStatementResponse>;
+    listSparkBatchJobs(options?: ListSparkBatchJobsOptions): Promise<ListSparkBatchJobsResponse>;
```

Comment:
the list methods should be returning PagedAsyncIterableIterators:

https://azure.github.io/azure-sdk/typescript_design.html#ts-pagination

This custom pagination will need to be wrapped.

Reply:
Got it, that is due to our swagger missing pageable extension. I will discuss with @idear1203 how to fix it, update swagger or just wrap custom code at TS.

---
In reply to: [521606981](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521606981) [](ancestors = 521606981)

Reply:
HI @xirzec , we plan to support it from swagger, so can we just keep it at this PR, becauze we need to do board review by this month, that may cause big change of swagger.
I also want to confirm with which is should be set first priority, currrent review or board review?

 I found release related info from [https://azure.github.io/azure-sdk/policies_reviewprocess.html#api-listings](https://azure.github.io/azure-sdk/policies_reviewprocess.html#api-listings)

---
In reply to: [521850566](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521850566) [](ancestors = 521850566,521606981)

Reply:
One small note that APIView now supports JS, so that guidance is a bit of date.

You don't necessary need to change the swagger in order to make this pageable, as you can do it by hand in the convenience layer. This is what other packages do today since our code generator doesn't even support generating pageables, though @joheredi is working on that.

Reply:
To answer your other question, if you want to do an API review with the board (the step after the service informational), then you should have packages checked in for all languages (though not published yet), APIView links for each language, and also code samples for each language. Sometimes we don't get through all 4 languages in one meeting, so it might take two.

One other note is that even if you get your packages merged and your API shape blessed by the board this month, you won't be able to ship any packages until January.